gtsam/sfm/ShonanAveraging.cpp

```diff
@@ -471,57 +471,86 @@ Sparse ShonanAveraging<d>::computeA(const Values &values) const {
   return Lambda - Q_;
 }
 
+/* ************************************************************************* */
+template <size_t d>
+Sparse ShonanAveraging<d>::computeA(const Matrix &S) const {
+  auto Lambda = computeLambda(S);
+  return Lambda - Q_;
+}
+
+/* ************************************************************************* */
+// Perturb the initial initialVector by adding a spherically-uniformly
+// distributed random vector with 0.03*||initialVector||_2 magnitude to
+// initialVector
+// ref : Part III. C, Rosen, D. and Carlone, L., 2017, September. Computational
+// enhancements for certifiably correct SLAM. In Proceedings of the
+// International Conference on Intelligent Robots and Systems.
+Vector perturb(const Vector &initialVector) {
```

Comment:
This looks good, but I think it should also be ```const```?

Reply:
Non-member function can't add const

Reply:
Actually, if it's a local function this can just be made static in this file.

Reply:
OK, I don't really have strong feelings about this one way or the other -- happy to resolve this as-is if you are @dellaert?